src/main/java/dev/jbang/util/ModuleUtil.java

```diff
@@ -1,13 +1,37 @@
 package dev.jbang.util;
 
+import java.io.InputStream;
+import java.net.URL;
 import java.nio.file.Path;
 
+import javax.annotation.Nullable;
+
+import dev.jbang.catalog.CatalogUtil;
 import dev.jbang.source.Project;
 
 /**
  * WARNING: This file MUST be compiled with Java 9+
  */
 public class ModuleUtil {
+	public static boolean isModule(Path file) {
+		try {
+			URL url = new URL("jar:" + file.toUri().toURL() + "!/module-info.class");
+			try (InputStream s = url.openStream()) {
+				return true;
+			}
+		} catch (Exception ex) {
+		}
+		try {
+			// TODO This is a very specific test, we should do better
+			URL url = new URL("jar:" + file.toUri().toURL() + "!/META-INF/versions/9/module-info.class");
```

Comment:
what is this trying to do ? detect multi-jarred modules?

Reply:
Not multi-jarred, but multi-versioned. JARs that can run on, for example, Java 8 and Java 9 but have slightly different code for each. In this case I've found JARs that have everything compiled with Java 8 but have a single module-info.class file compiled with Java 9